ext/mysqlnd/mysqlnd_result.c

```diff
@@ -302,7 +302,7 @@ mysqlnd_query_read_result_set_header(MYSQLND_CONN_DATA * conn, MYSQLND_STMT * s)
 				if (FAIL == (ret = result->m.read_result_metadata(result, conn))) {
 					/* For PS, we leave them in Prepared state */
 					if (!stmt && conn->current_result) {
-						mnd_efree(conn->current_result);
+						result->m.free_result(result, TRUE);
```

Comment:
Then I'd rather do
```suggestion
						conn->current_result->m.free_result(onn->current_result, TRUE);
```
or maybe just `ZEND_ASSERT(result == conn->current_result);` before the `if` statement, and use `result` within the `if` body throughout.

This looks all a bit messed up due to the aliasing, and its inconsistent usage; see also the following `if` statement.

Reply:
Yes, that's what I meant.

The assert is not necessary. But for consistency whenever we are inside `if(! stmt)` guard then let's use `conn->current_result`. See line 266